frame/pablo/src/lib.rs

```diff
@@ -420,6 +440,34 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn enable_twap(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			let _who = ensure_signed(origin)?;
```

Comment:
What is the rationale behind allowing any signed origin to enable this?

Reply:
Yes, I think we should only allow Root or Pool creator to enable this.